honeybee_radiance_command/gensky.py

```diff
@@ -0,0 +1,257 @@
+        """time."""
+        return self._time
+
+    @time.setter
+    def time(self, value):
+        if not value:
+            self._time = None
+
+        elif isinstance(value, float):
+            hour, minute = str(value).split('.')
+
+            # Validate hour
+            hour = int_in_range(int(hour), 0, 23)
+
+            # Validate minute
+            if minute in ('25', '5', '75'):
+                minutes = {'25': '15', '5': '30', '75': '45'}
+
+                self._time = '%s:%s' % (hour, minutes[minute])
+            else:
+                raise ValueError(
+                    'Decimal values are only allowed for 15 minute increments.'
+                    ' Such as 9.25, 9.5, and 9.75 which will become 9:15, 9:30 and'
+                    ' 9:45 respectively. You provided %s' % (value)
+                )
+
+        elif isinstance(value, str) and ':' in value:
+            if ':' in value:
+                hour, minute = value.split(':')
+
+            # Validate hour
+            hour = int_in_range(int(hour), 0, 23)
+            # Validate minute
+            minute = int_in_range(int(minute), 0, 59)
+
+            self._time = value
+
+        else:
+            raise ValueError(
+                '%s is not a valid format. Examples of acceptable formats are'
+                ' a float value of 21.5 or a string value of 21:30' % (value)
+            )
+
+
+    @property
+    def time_zone(self):
+        """time zone."""
+        return self._time_zone
+
+    @time_zone.setter
+    def time_zone(self, value):
+        time_zones = [
+            'YST', 'PST', 'MST', 'CST', 'EST', 'GMT', 'CET', 'EET', 'AST', 'GST', 'IST',
+            'JST', 'NZST', 'YDT', 'PDT', 'MDT', 'CDT', 'EDT', 'BST', 'CEST', 'EEST',
+            'ADT', 'GDT', 'IDT', 'JDT', 'NZDT']
+        if not value:
+            self._time_zone = None
+        elif value.upper() in time_zones:
+            self._time_zone = value
+        else:
+            raise ValueError(
+                'Time zone must a three letter string from the following'
+                ' options %s.' % (time_zones)
+            )
+
+    @property
+    def solar_time(self):
+        """Whether solar time is requested."""
+        return self._solar_time
+
+    @solar_time.setter
+    def solar_time(self, value):
+        if not value:
+            self._solar_time = None
+        elif isinstance(value, bool):
+            self._solar_time = value
+        else:
+            raise ValueError('Solar time only accepts True or False as a value.')
+
+    @property
+    def input(self):
+        """Input string for gensky."""
+
+        if not self.month and not self.day and not self.time:
+            self._input = None
+            return self._input
+
+        # Only month, day and time provided as arguments
+        elif not (self.time_zone and self.solar_time):
+            self._input = '%s %s %s' % (str(self.month), str(self.day), str(self.time))
+            return self._input
```

Comment:
I might be wrong but this logic looks flawed to me. Think about a case that month is provided but not day or time. It will not meet the initial condition and then comes here and the condition is correct which is not what you want.

What you can do to simplify the process is to create a base string and then add extra strings to them if available.

Reply:
month day and time are mandatory inputs or you can use the -ang option with altitude and azimuth. So if only month is provided, the [validate](https://github.com/ladybug-tools/honeybee-radiance-command/pull/154/files#diff-429d0b7200e14c15d550971eb667b6e52b5c5d6d135500ae4c132351b334484aR250) method will raise an exception.

Reply:
I see. So the assumption is that this case never happens. I'm cool with that.